packages/ember/tests/component_registration_test.js

```diff
@@ -1,200 +1,172 @@
+  ['@test Late-registered components can be rendered with ONLY the template registered on the container'](assert) {
+    this.addTemplate('application', '<div id=\'wrapper\'>hello world {{borf-snorlax}}-{{#borf-snorlax}}!!!{{/borf-snorlax}}</div>');
 
-  boot(() => {
-    appInstance.register('template:components/sally-rutherford', compile('funkytowny{{yield}}'));
-    appInstance.register('component:sally-rutherford', Component);
-  });
+    this.addComponent('borf-snorlax', {
+      template: 'goodfreakingTIMES{{yield}}'
+    });
 
-  equal(jQuery('#wrapper').text(), 'hello world funkytowny-funkytowny!!!', 'The component is composed correctly');
-});
+    this.visit('/').then(() => {
+      let text = this.$('#wrapper').text().trim();
+      assert.equal(text, 'hello world goodfreakingTIMES-goodfreakingTIMES!!!', 'The component is composed correctly');
+    });
+  }
 
-QUnit.test('Late-registered components can be rendered with ONLY the template registered on the container', function() {
-  setTemplate('application', compile('<div id=\'wrapper\'>hello world {{borf-snorlax}}-{{#borf-snorlax}}!!!{{/borf-snorlax}}</div>'));
+  ['@test Assigning layoutName to a component should setup the template as a layout'](assert) {
+    assert.expect(1);
 
-  boot(() => {
-    appInstance.register('template:components/borf-snorlax', compile('goodfreakingTIMES{{yield}}'));
-  });
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
+    this.addTemplate('foo-bar-baz', '{{text}}-{{yield}}');
+    this.add('controller:application', Controller.extend({
+      text: 'outer'
+    }));
 
-  equal(jQuery('#wrapper').text(), 'hello world goodfreakingTIMES-goodfreakingTIMES!!!', 'The component is composed correctly');
-});
+    this.addComponent('my-component', {
+      ComponentClass: Component.extend({
+        text: 'inner',
+        layoutName: 'foo-bar-baz'
+      })
+    });
 
-QUnit.test('Assigning layoutName to a component should setup the template as a layout', function() {
-  expect(1);
+    this.visit('/').then(() => {
+      let text = this.$('#wrapper').text().trim();
+      assert.equal(text, 'inner-outer', 'The component is composed correctly');
+    });
+  }
 
-  setTemplate('application', compile('<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>'));
-  setTemplate('foo-bar-baz', compile('{{text}}-{{yield}}'));
+  ['@test Assigning layoutName and layout to a component should use the `layout` value'](assert) {
+    assert.expect(1);
 
-  boot(() => {
-    appInstance.register('controller:application', Controller.extend({
-      'text': 'outer'
+    this.addTemplate('application', '<div id=\'wrapper\'>{{#my-component}}{{text}}{{/my-component}}</div>');
+    this.addTemplate('foo-bar-baz', 'No way!');
+    this.add('controller:application', Controller.extend({
```

Comment:
per the original test, this should be registered onto an autobooted application instance.

the `my-component` below should be as well.